boring/src/x509/crl.rs

```diff
@@ -1,3 +1,45 @@
+//! Certificate revocation lists describe certificates that have been revoked
+//! by their issuer and should no longer be trusted.
+//!
+//! An `X509CRL` can be provided along with an issuing `X509` to verify that
+//! issued certificates have not been revoked.
+//!
+//! # Example
+//!
+//! ```rust
+//! use boring::hash::MessageDigest;
+//! use boring::pkey::{PKey, Private};
+//! use boring::x509::crl::{X509CRLBuilder, X509Revoked};
+//! use boring::x509::extension::BasicConstraints;
+//! use boring::x509::verify::X509VerifyFlags;
+//! use boring::x509::X509Extension;
+//! use boring::x509::X509;
+//! use boring::x509::store::{X509Store, X509StoreBuilder};
+//! use boring::asn1::Asn1Time;
+//! use boring::bn::BigNum;
+//! use boring::error::ErrorStack;
+//!
+//! fn crl_checking_store(issuer: X509, pkey: PKey<Private>) -> Result<X509Store, ErrorStack> {
+//!    let mut builder = X509CRLBuilder::new()?;
+//!    builder.set_issuer_name(issuer.subject_name())?;
+//!    builder.add_revoked(X509Revoked::from_parts(
+//!        &*BigNum::from_u32(1)?.to_asn1_integer()?,
+//!        &*Asn1Time::days_from_now(0)?
```

Comment:
Nitpick: I *suspect* the `&*` is not actually required because of [Deref coercion](https://doc.rust-lang.org/std/ops/trait.Deref.html#more-on-deref-coercion); just `&` would work. (Like it does when passing `&vec` where `&[u8]` is expected.)

Reply:
Hm `&` doesn't work and I cannot say I understand why. Most surprising to me:

```
error[E0308]: `?` operator has incompatible types
  --> src/x509/crl.rs:25:28
   |
19 |    let t : &Asn1TimeRef = &Asn1Time::days_from_now(0)?;
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Asn1TimeRef`, found struct `Asn1Time`
   |
   = note: `?` operator cannot convert from `Asn1Time` to `Asn1TimeRef`
```

but unwrap instead of `?` compiles fine!
```
let t : &Asn1TimeRef = &Asn1Time::days_from_now(0).unwrap();
```

`&&` works but I don't know if that's actually clearer than `&*`

Reply:
Ah :-( `&(Asn1Time::days_from_now(0)?)` would probably work but that's definitely uglier.

Reply:
…wait, that doesn't make sense, that's the default. Hm, let's try this in a playground…

Reply:
I'm confused, it seems to work there: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cabc3592e5919daa8903819ea8589e4f

Reply:
…but fails with custom types. Gotta figure out what's different. https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e4f4513f6c6f7c4f92df4cde94dc5292

Reply:
:-/ Apparently unsized types (like slices) get special treatment. I'll ask about this on the Rust user forums, but for now what you have is probably best!

Reply:
https://users.rust-lang.org/t/deref-coercion-behaves-differently-for-sized-vs-sized-targets/77774